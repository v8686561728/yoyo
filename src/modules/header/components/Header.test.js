import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Header } from "./Header";
import { mapStateToProps, mapDispatchToProps } from "./Header";

describe("Header", () => {
  let props, props2, wrapper;
  beforeEach(() => {
    props = {
      classes: "root",
      isLoggedIn: true,
      userDetails: { first_name: "yudha" }
    };
    props2 = {
      classes: "root",
      isLoggedIn: false,
      userDetails: { first_name: "yudha" }
    };
  });

  it("should render Header component", () => {
    wrapper = shallow(<Header {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  //myProfile method
  // it("should trigger myProfile", () => {
  //   wrapper = shallow(<Header {...props} />);
  //   console.log(shallowToJson(wrapper));
  // });

  it("should call mapDispatchToProps", () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).logout();
  });

  //login
  it("should trigger logout click event", () => {
    wrapper = shallow(<Header logout={jest.fn()} {...props2} />);
    wrapper.find("#authButton").simulate("click");
  });

  //logout
  it("should trigger logout click event", () => {
    wrapper = shallow(<Header logout={jest.fn()} {...props} />);
    wrapper.find("#authButton").simulate("click");
  });

  it("should trigger myProfile  click event", () => {
    wrapper = shallow(<Header {...props} />);
    wrapper.find("#profile").simulate("click");
  });

  it("should trigger sendGiftButton  click event", () => {
    wrapper = shallow(<Header {...props} />);
    wrapper.find("#sendGiftButton").simulate("click");
  });

  it("should trigger giftReceiveButton  click event", () => {
    wrapper = shallow(<Header {...props} />);
    wrapper.find("#giftReceiveButton").simulate("click");
  });

  it("should trigger home  click event", () => {
    wrapper = shallow(<Header {...props} />);
    wrapper.find("#home").simulate("click");
  });

  it("should map state to props when user is not logged in", () => {
    const initialState = {
      login: {
        loginStatus: true,
        detailsObject: {}
      }
      //   userDetails: {
      //     first_name: "John"
      //   }
    };
    expect(mapStateToProps(initialState).isLoggedIn).toEqual(true);
  });
});
