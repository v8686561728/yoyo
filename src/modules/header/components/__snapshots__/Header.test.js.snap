// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h6"
      >
        <WithStyles(ForwardRef(Button))
          id="home"
          onClick={[Function]}
        >
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.2em",
              }
            }
          >
            YOYOGift
          </span>
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        className="headerButton"
        color="inherit"
        id="giftReceiveButton"
        onClick={[Function]}
      >
        GIFTS RECEIVED
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="headerButton"
        color="inherit"
        id="sendGiftButton"
        onClick={[Function]}
      >
        GIFTS SENT
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="headerButton"
        color="inherit"
        id="profile"
        onClick={[Function]}
      >
        yudha
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="headerButton"
        color="inherit"
        id="authButton"
        onClick={[Function]}
      >
        LOGOUT
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
